src/intrinsics/ecma262/Object.js

```diff
@@ -118,8 +130,26 @@ export default function(realm: Realm): NativeFunctionValue {
       }
     }
 
+    if (to_must_be_abstract) {
+      let types = new TypesDomain(FunctionValue);
+      let values = new ValuesDomain();
+      invariant(realm.generator);
+      to = realm.generator.derive(types, values, [target, ...sources], _args => {
+        return t.callExpression(
+          t.memberExpression(t.identifier("Object"), t.identifier("assign")),
+          ((_args: any): Array<any>)
+        );
+      });
+      target.makePartial();
+      target.makeSimple();
+      // Make AbstractObjectValue as Object.assign doesn't copy getters/setter
+      invariant(to instanceof AbstractObjectValue);
+      to.makeSimple();
```

Comment:
I don't think we know this is safe. AFAIK there's nothing that tells us `to` is simple.

Reply:
If we can determine this, that will make sense, otherwise tests fail without this.